### DISCUSSION TITLE
Q&A Engineering #26: How does pseudo-labeling on test set predictions improve semi-supervised models?

### DISCUSSION THREAD

**Discussion by BerattCelikk** [BerattCelikk/daily-engineering-log#388] [Q&A] [answered]
How does pseudo-labeling on test set predictions improve semi-supervised models?

**Comment by BerattCelikk** [comment, marked as answer]
Selecting high-confidence test predictions (e.g. p > 0.98 or p < 0.02) and augmenting training data shifts decision boundaries towards lower-density regions in the feature space, improving margin separation.
